lace/lace_codebook/src/value_map.rs

```diff
@@ -0,0 +1,368 @@
+                1 => Category::Bool(true),
+                _ => panic!("{ix} is too large for boolean map"),
+            },
+        }
+    }
+
+    pub fn contains_cat(&self, cat: &Category) -> bool {
+        match (self, cat) {
+            (Self::String(map), Category::String(x)) => map.contains_cat(x),
+            (Self::U8(k), Category::U8(x)) => (*x as usize) < *k,
+            (Self::Bool, Category::Bool(_)) => true,
+            _ => false,
+        }
+    }
+
+    pub fn iter(&self) -> CategoryIter {
+        CategoryIter::new(self)
+    }
+
+    /// Determine whether a value map is an extended version of another
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use lace_codebook::ValueMap;
+    /// use std::collections::BTreeSet;
+    /// use lace_data::Category;
+    ///
+    /// let mut cats: BTreeSet<String> = BTreeSet::new();
+    ///
+    /// cats.insert("B".into());
+    /// cats.insert("C".into());
+    /// cats.insert("A".into());
+    ///
+    /// let value_map_1 = ValueMap::new(cats.clone());
+    ///
+    /// assert!(value_map_1.is_extended(&value_map_1));
+    ///
+    /// cats.insert("D".into());
+    ///
+    /// let value_map_2 = ValueMap::new(cats);
+    ///
+    /// assert!(value_map_1.len() < value_map_2.len());
+    /// assert!(value_map_1.is_extended(&value_map_2));
+    /// assert!(!value_map_2.is_extended(&value_map_1));
+    /// ```
+    ///
+    /// Integer valuemap
+    ///
+    /// ```
+    /// # use lace_codebook::ValueMap;
+    /// let value_map_2 = ValueMap::U8(2);
+    /// let value_map_3 = ValueMap::U8(3);
+    /// let value_map_4 = ValueMap::U8(4);
+    ///
+    /// assert!(value_map_2.is_extended(&value_map_2));
+    /// assert!(value_map_2.is_extended(&value_map_3));
+    /// assert!(value_map_2.is_extended(&value_map_4));
+    ///
+    /// assert!(!value_map_3.is_extended(&value_map_2));
+    /// assert!(value_map_3.is_extended(&value_map_3));
+    /// assert!(value_map_3.is_extended(&value_map_4));
+    ///
+    /// assert!(!value_map_4.is_extended(&value_map_2));
+    /// assert!(!value_map_4.is_extended(&value_map_3));
+    /// assert!(value_map_4.is_extended(&value_map_4));
+    /// ```
+    ///
+    /// ```
+    /// # use lace_codebook::ValueMap;
+    /// let value_map = ValueMap::Bool;
+    ///
+    /// assert!(value_map.is_extended(&value_map));
+    /// ```
+    pub fn is_extended(&self, other: &Self) -> bool {
+        // TODO: DRY arms!
+        match (self, other) {
+            (Self::String(a), Self::String(b)) => {
+                if b.len() < a.len() {
+                    return false;
+                }
+                a.to_cat
+                    .iter()
+                    .zip(b.to_cat.iter())
+                    .all(|(ai, bi)| ai == bi)
+            }
+            (Self::U8(k_a), Self::U8(k_b)) => k_b >= k_a,
+            (Self::Bool, Self::Bool) => true,
+            _ => false,
+        }
+    }
+}
+
+macro_rules! map_try_from_vec {
```

Comment:
Is it easier/clearer to have a macro than to just write it out, given that (AFAICT) this macro is only used once?

I see that it used to be used more than once but isn't now; I don't mind leaving it as is rather than trying to change it, if it works.